src/masternodeman.cpp

```diff
@@ -856,8 +857,7 @@ void CMasternodeMan::CheckSameAddr()
                     pverifiedMasternode = pmn;
                 }
             } else {
-                // clear the reference
-                pverifiedMasternode = NULL;
+                pverifiedMasternode = pmn->IsPoSeVerified() ? pmn : NULL;
             }
             pprevMasternode = pmn;
         }
```

Comment:
Yes, it doesn't ensure that for one single run but it does so after few runs. And no, simpler version will ban everyone but the first node which is not necessary the one that is/will be verified. At start no one is verified actually, every node slowly gaining its score, so you'll just ban random nodes that way.